pyproject.toml

```diff
@@ -1,61 +1,75 @@
-"write_to" = "geoips_clavrx/_version.py"  # Write an actual file
+# NOTE: You must CREATE a _version.py file and commit it via git!
+#     That file will have 0.0.0 as the version - which will get updated when
+#     pip installed.
+# FURTHER NOTE: pip install -e will actually NOT update _version.py.
+#     if you must have a valid version in your local editable install (which
+#     is typically not necessary), you can manually update _version.py
+#     with the following steps:
+#     > pip install poetry-dynamic-versioning
+#     > poetry-dynamic-versioning
+#     (those steps are called automatically when running pip install, but
+#     NOT when running pip install -e)
+[tool.poetry-dynamic-versioning.substitution]
+files = ["*/_version.py"]
 
-[project]
-name = "geoips_clavrx"
-authors = []
-description = "CLAVR-x GeoIPS Plugin Package"
-readme = "README.md"
-requires-python = ">=3.9"
-keywords = []
-license = {file = "LICENSE"}
-classifiers = [
-    "Programming Language :: Python :: 3",
+# This is where you would specify a pre-build script from pyproject.toml/poetry.
+# [tool.poetry.build]
+# generate-setup-file = false  # I have no idea what this does
+# script = prebuild.py  # This gets called as a python script prior to running build steps
+
+[build-system]
+requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning>=1.0.0,<2.0.0"]
+build-backend = "poetry_dynamic_versioning.backend"                          # object used to perform the build process
+
+[tool.poetry.dependencies] # must download to run
+python = ">=3.9"       # mandatory to declare the required python version
+
+# Optional dependencies
+# To add a new optional dependency it must be added both here and in one of the "extra"
+# groups below.
+#
+# Build group
+requests = { version = "*", optional = true }
+
+# NOTE system libjpeg also required, via conda
+[tool.poetry.extras]
+build = [
+    "requests",
 ]
-dynamic = ["version"]
-
-[project.entry-points."geoips.plugin_packages"]
-"geoips_clavrx" = "geoips_clavrx"
-
-[project.entry-points."geoips.readers"]
-clavrx_hdf4 = "geoips_clavrx.plugins.modules.readers.clavrx_hdf4"
-
-[project.entry-points."geoips.colormappers"]
-cmap_cldFraction = "geoips_clavrx.plugins.modules.colormappers.cmap_cldFraction"
-cmap_cldType = "geoips_clavrx.plugins.modules.colormappers.cmap_cldType"
-cmap_cldMask = "geoips_clavrx.plugins.modules.colormappers.cmap_cldMask"
-cmap_cldHeight = "geoips_clavrx.plugins.modules.colormappers.cmap_cldHeight"
-cmap_cldOpd = "geoips_clavrx.plugins.modules.colormappers.cmap_cldOpd"
-cmap_cldPhase = "geoips_clavrx.plugins.modules.colormappers.cmap_cldPhase"
-cmap_cldTemp = "geoips_clavrx.plugins.modules.colormappers.cmap_cldTemp"
-cmap_IR = "geoips_clavrx.plugins.modules.colormappers.cmap_IR"
-cmap_cldReff = "geoips_clavrx.plugins.modules.colormappers.cmap_cldReff"
+
+[tool.poetry.plugins."geoips.plugin_packages"]
+geoips = "geoips_clavrx"
```

Comment:
I think this is wrong - shouldn't it be geoips_clavrx = "geoips_clavrx"

Reply:
Yes, you are correct. I just corrected this.